app/controllers/users_controller.rb

```diff
@@ -1,3 +1,27 @@
 class UsersController < ApplicationController
-  def new; end
+  attr_reader :user
+  def show
+    @user = User.find params[:id]
```

Comment:
dòng lệnh trên thực hiện công việc gì?
có những cách nào làm việc đấy?
và theo bạn thì nên dùng cách nào?

Reply:
### User.find sẽ tìm trong database record có trường id = params :id truyền vào. sau đó gán cho instance @user
### Các cách để find user:
+ find (*arg) -> tìm theo id, nếu ko có raise RecordNotFound exception
+ find_by (arg,*arg) -> tìm theo 1 field trong db , nếu thấy return 1st record, nếu không có return nil
+ find_by! (arg,*arg) -> tương tự find_by, nếu không có raise RecordNotFound exception
### Theo em nghĩ nên dùng find_by vì có thể tìm trên nhiều trường, nếu chỉ tìm theo id thì dùng find không thôi.
Ngoài ra nếu xử lý not found,  nil dễ xử lý hơn so với xử lý exception phải dùng begin/ rescue.